public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { DataSourcePluginOptionsEditorProps, DataSourceSettings, SelectableValue } from '@grafana/data';
+import { DataSourceHttpSettings, FormLabel, Input, SecretFormField, Select } from '@grafana/ui';
+import { InfluxOptions, InfluxSecureJsonData } from '../types';
+
+const httpModes = [{ label: 'GET', value: 'GET' }, { label: 'POST', value: 'POST' }];
+
+export type Props = DataSourcePluginOptionsEditorProps<InfluxOptions>;
+
+type InfluxDataSourceSettings = DataSourceSettings<InfluxOptions, InfluxSecureJsonData>;
+
+export interface State {
+  config: InfluxDataSourceSettings;
+}
```

Comment:
I've seen this pattern a lot in our code and in candidates code during technical exercise. 

By adding `config` as an item to state instead of all it's individual properties means that any changes to that item will affect and re render all components bound to that item. I know this is not a performance intense part of the application but I think that we should try to avoid this pattern when possible.

You should declare state as follows and change the rest of the Component accordingly:
```
export interface State extends InfluxDataSourceSettings {}
```